chips/stm32f429zi/src/ethernet/ethernet.rs

```diff
@@ -0,0 +1,2288 @@
+                return Ok(());
+            }
+        }
+
+        Err(ErrorCode::BUSY)
+    }
+
+    #[allow(dead_code)]
+    // When a standard HIL will be implemented, this method will be used
+    fn disable_receiver(&self) -> Result<(), ErrorCode> {
+        self.disable_dma_reception()?;
+        self.disable_mac_receiver();
+
+        Ok(())
+    }
+
+    fn is_reception_enabled(&self) -> bool {
+        self.is_mac_receiver_enabled() && self.is_dma_reception_enabled()
+    }
+
+    fn enable_all_normal_interrupts(&self) {
+        self.enable_normal_interrupts();
+        self.enable_early_receive_interrupt();
+        self.enable_receive_interrupt();
+        self.enable_transmit_buffer_unavailable_interrupt();
+        self.enable_transmit_interrupt();
+    }
+
+    fn enable_all_error_interrupts(&self) {
+        self.enable_abnormal_interrupt_summary();
+        self.enable_early_receive_interrupt();
+        self.enable_fatal_bus_error_interrupt();
+        self.enable_early_transmit_interrupt();
+        self.enable_receive_watchdog_timeout_interrupt();
+        self.enable_receive_process_stopped_interrupt();
+        self.enable_receive_buffer_unavailable();
+        self.enable_underflow_interrupt();
+        self.enable_overflow_interrupt();
+        self.enable_transmit_jabber_interrupt();
+        self.enable_transmit_process_stopped_interrupt();
+    }
+
+    fn enable_all_interrupts(&self) {
+        self.enable_all_normal_interrupts();
+        self.enable_all_error_interrupts();
+    }
+
+    fn handle_normal_interrupt(&self) {
+        if self.did_transmit_interrupt_occur() {
+            self.clear_transmit_interrupt();
+            self.client.map(|client| {
+                client.tx_done(
+                    Ok(()),
+                    self.transmit_packet.take().unwrap(),
+                    self.transmit_packet_length.take().unwrap(),
+                    self.packet_identifier.take().unwrap(),
+                    None,
+                )
+            });
+        } else if self.did_transmit_buffer_unavailable_interrupt_occur() {
+            self.clear_transmit_buffer_unavailable_interrupt();
+        } else if self.did_receive_interrupt_occur() {
+            self.clear_receive_interrupt();
+            self.client.map(|client| {
+                let received_packet = self.received_packet.take().unwrap();
+                client.rx_packet(received_packet, None);
+                self.received_packet.put(Some(received_packet));
+            });
+            // Receive the following packet
+            assert_eq!(Ok(()), self.receive_packet());
+        } else if self.did_early_receive_interrupt_occur() {
+            self.clear_early_receive_interrupt();
+        }
+    }
+
+    fn handle_abnormal_interrupt(&self) {
+        if self.did_fatal_bus_error_interrupt_occur() {
+            self.clear_fatal_bus_error_interrupt();
+            panic!("Fatal bus error");
```

Comment:
Perhaps we should avoid panicing here in the long term, except if these are errors which really put the chip in an unrecoverable state. However, something like a "transmit buffer underflow" seems like the kind of interrupt which could be occurring at runtime without major impact (except for a broken Ethernet frame), depending on the AHB bus.

Reply:
Removed most of the panics and added additionally comments to explain the intent of the panic handler actions. Might require future revision.